packages-ts/gauntlet-terra-contracts/src/commands/contracts/ocr2/inspection/responses.ts

```diff
@@ -0,0 +1,195 @@
+  const operators = rdd.operators
+  const aggregatorProps = rdd.contracts[args[0]]
+  const aggregatorOracles = aggregatorProps['oracles']
+  const aggregatorOperators: string[] = aggregatorOracles.map((o) => o.operator)
+  const transmitters = aggregatorOperators.map((o) => rdd.operators[o].ocrNodeAddress[0])
+
+  const observersInfo: { [address: string]: ObserverInfo } = {}
+  aggregatorOracles.forEach((oracle) => {
+    observersInfo[operators[oracle.operator].ocrNodeAddress[0]] = {
+      name: oracle.operator,
+      apis: oracle.api,
+    }
+  })
+
+  return {
+    transmitters,
+    description: aggregatorProps.name,
+    observersInfo,
+  }
+}
+
+const makeOnchainData = (provider: LCDClient) => async (
+  instructionsData: any[],
+  input: CommandInput,
+  aggregator: string,
+): Promise<ContractExpectedInfo> => {
+  const latestRoundData = instructionsData[0]
+  const transmitters = instructionsData[1]
+  const description = instructionsData[2]
+
+  let event
+  const onchainEvent = await getLatestOCRNewTransmissionEvent(provider, aggregator)
+  if (onchainEvent) {
+    event = {
+      answer: Number(onchainEvent.answer[0]),
+      transmitter: onchainEvent.transmitter[0],
+      observations: onchainEvent.observations,
+      observers: onchainEvent.observers[0],
+      observations_timestamp: Number(onchainEvent.observations_timestamp[0]),
+      aggregator_round_id: Number(onchainEvent.aggregator_round_id[0]),
+    }
+  }
+
+  return {
+    transmitters: transmitters.addresses,
+    description,
+    aggregatorAddress: aggregator,
+    latestRoundData: {
+      roundId: latestRoundData.round_id,
+      answer: latestRoundData.answer,
+      observationsTimestamp: latestRoundData.observations_timestamp,
+      transmissionTimestamp: latestRoundData.transmission_timestamp,
+    },
+    event: event,
+  }
+}
+
+const inspect = (inputData: CommandInput, onchainData: ContractExpectedInfo): boolean => {
+  let inspections: inspection.Inspection[] = [
+    inspection.makeInspection(onchainData.transmitters, inputData.transmitters, 'Transmitters'),
+    inspection.makeInspection(onchainData.description, inputData.description, 'Description'),
+  ]
+  inspection.inspect(inspections)
+
+  const event = onchainData.event
+  if (!event) {
+    logger.error(
+      `Could not find NewTransmission event of aggregator ${onchainData.aggregatorAddress} for the latest round).`,
+    )
+    return false
+  }
+  logger.debug(event)
+  logger.debug(onchainData.transmitters)
+
+  const roundTime = new Date(Number(event.observations_timestamp) * 1000)
+  logger.info(`Found NewTransmission event for round ${event.aggregator_round_id} (${roundTime})`)
+
+  let responsiveNodes: { [address: string]: ObserverInfo } = {}
+  let unresponsiveNodes: { [address: string]: ObserverInfo } = {}
+
+  const parsedObserversIndexes = parseObserversByLength(event.observers, event.observations.length)
+
+  parsedObserversIndexes.forEach((val, i) => {
```

Comment:
This and [next](inspection.inspect(inspections)) are hard to understand. 
- Use a simpler data structure. Maybe only have one `responsiveNodes: Oracle[]`
- `unresponsiveNodes` could just be `unresponsiveNodes = input.oracles.filter((o) => !responsiveNodes.includes(o))`
- Try to declare the variables along to their values, that way we avoid mutability and increase readability 
  - `const responsiveNodes = parsedObserversIndexes.map(() => ...)`

Reply:
I initially had implemented it as you describe above, but I noticed there was too much searching in lists(filter,map, find). Although the size is too small to even be significant for performance,  I tried to have a general structure from which you can directly find elements' presence. This is also why for the CommandInput type I opted for a dictionary rather than a single Observer/Oracle object.
I agree though that readability took a hit with that.